unittests/platformagnostictests.py

```diff
@@ -294,3 +296,102 @@ def test_setup_with_unknown_option(self):
         for option in ('not_an_option', 'b_not_an_option'):
             out = self.init(testdir, extra_args=['--wipe', f'-D{option}=1'], allow_fail=True)
             self.assertIn(f'ERROR: Unknown options: "{option}"', out)
+
+    @expectedFailure
+    def test_configure_new_option(self) -> None:
+        """Adding a new option without reconfiguring should work."""
+        testdir = self.copy_srcdir(os.path.join(self.common_test_dir, '40 options'))
+        self.init(testdir)
+        with open(os.path.join(testdir, 'meson_options.txt'), 'a', encoding='utf-8') as f:
```

Comment:
Maybe we should teach the test infra to use meson.options from now on. Work for later. No action on your part.

Reply:
Probably. But also I wanted to be sure that I was testing converting between meson.options and meson_options.txt